### PR TITLE
Allow methods with empty return type to be callable.

### PR DESCRIPTION
**Pull request by nlordell**
This PR allows callable methods have an empty return type to be callable.

### Test Plan

Added empty return type view method to example.

### CODE REVIEW

File: src/contract/method.rs [hunk 1]
```diff
@@ -16,9 +16,64 @@ use std::pin::Pin;
...
+        }
+
+        Ok(())
+    }
+}
+
+impl<T: Detokenize> Detokenizable for T {
+    type Output = Self;
+
+    fn is_empty() -> bool {
```
Comment (fleupold) [inline src/contract/method.rs]: I find this name a little bit misleading as e.g. a string or vector could also be empty. Is this really just to see if this type can be used to symbolize an error? Maybe we could just call it is Void (or have a more dynamic way of downcasting the concrete type into void)?

Which makes me wonder what happens in geth when we revert without a message on a void function. Is it distinguishable from a success?

**Review by fleupold — APPROVED** [review]
Nice, thanks for the quick fix. Just a minor concern about the naming of is_empty, but nothing blocking.

File: src/contract/method.rs (hunk 1 above)
Comment (nlordell) [reply src/contract/method.rs]: > Maybe we could just call it is Void

I like `is_void`.

> Which makes me wonder what happens in geth when we revert without a message on a void function. Is it distinguishable from a success?

No, and that is sadly why we need it :(

**Review by e00E — APPROVED** [review]

File: src/contract/method.rs (hunk 1 above)
Comment (nlordell) [reply src/contract/method.rs]: I renamed the method to `is_void` and added a note in the comments about how it cannot be distringuised from an `INVALID` opcode or `REVERT` without data.

**Comment by bh2smith** [comment]
Was there a motivation in mind for this improvement?

**Comment by nlordell** [comment]
@bh2smith It was more of a bug fix, the generated code was having issues with the `BatchExchangeViewer`

**Pull request merged**